worker/src/ocall_bridge/component_factory.rs

```diff
@@ -16,24 +16,61 @@
 
 */
 
+use crate::node_api_factory::NodeApiFactory;
 use crate::ocall_bridge::attestation_ocall_impl::RemoteAttestationOCallImpl;
-use crate::ocall_bridge::bridge_api::RemoteAttestationOCall;
+use crate::ocall_bridge::bridge_api::{RemoteAttestationOCall, WorkerOnChainOCall};
+use crate::ocall_bridge::on_chain_ocall_impl::WorkerOnChainOCallImpl;
+use crate::sync_block_gossiper::SyncBlockGossiper;
 use std::sync::Arc;
 
 /// Factory trait (abstract factory) that creates instances
 /// of all the components of the OCall Bridge
 pub trait OCallBridgeComponentFactory {
     /// remote attestation OCall API
     fn get_ra_api(&self) -> Arc<dyn RemoteAttestationOCall>;
+
+    /// on chain OCall API
+    fn get_oc_api(&self) -> Arc<dyn WorkerOnChainOCall>;
 }
 
 /// Concrete implementation, should be moved out of the OCall Bridge, into the worker
 /// since the OCall bridge itself should not know any concrete types to ensure
 /// our dependency graph is worker -> ocall bridge
-pub struct OCallBridgeComponentFactoryImpl;
+pub struct OCallBridgeComponentFactoryImpl<F, S>
+where
+    F: NodeApiFactory + 'static,
+    S: SyncBlockGossiper + 'static,
+{
+    node_api_factory: Arc<F>,
+    block_gossiper: Arc<S>,
+}
 
-impl OCallBridgeComponentFactory for OCallBridgeComponentFactoryImpl {
+impl<F, S> OCallBridgeComponentFactoryImpl<F, S>
+where
+    F: NodeApiFactory + 'static,
+    S: SyncBlockGossiper + 'static,
+{
+    pub fn new(node_api_factory: Arc<F>, block_gossiper: Arc<S>) -> Self {
+        OCallBridgeComponentFactoryImpl {
+            node_api_factory,
+            block_gossiper,
+        }
+    }
+}
+
+impl<F, S> OCallBridgeComponentFactory for OCallBridgeComponentFactoryImpl<F, S>
```

Comment:
Hmm, I suddenly have the impression that it is not good to consolidate add the APIs in a single construct. This only creates some layers of indirections. Why do you think this is better?